OpenTrons/opentrons-api
# refactor(app, api-client, react-api-client): unify analysis and run record for CommandText use

closes [AUTH-380](https://opentrons.atlassian.net/browse/AUTH-380)

# Overview

After a run is terminal, we access the run record rather than the most recently completed analysis in the command list in the run preview (or more aptly, run postview). This PR refactors the data sent to CommandText and ultimately many of its children and utilities to include only the necessary data from the run record if the run is terminal, or the most recent analysis if the run has not started or is in progress.

# Test Plan

- Run a protocol that loads at least one liquid (example linked as `rtp_tests.py` [here](https://opentrons.atlassian.net/browse/RQA-2645))
- Verify that run log before and during run shows full detail for load liquid command
- Cancel or finish run and verify that run log still shows full detail for load liquid command 

# Changelog

add new interface `CommandTextData` specifying the following subset of the union of `CompletedProtocolAnalysis` and `LegacyGoodRunData`
- pipettes
- labware
- modules
- liquids
- commands

implement this new interface in `CommandText`, its child components covering all possible command types, and its utilities that lookup entity data from their respective load commands

# Review requests

auth js

# Risk assessment

low-medium

[AUTH-380]: https://opentrons.atlassian.net/browse/AUTH-380?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
